### ISSUE TITLE
Consideration of additional pre-processing helper functions

### ISSUE THREAD

**Issue by medewitt** [epinowcast/epinowcast#404]
**Is your feature request related to a problem? Please describe.**
During the course of #402 we created a helper function for the test suite. 

```r
run_window_filter <- function(x, filter_report_remove = 10, filter_reference_include = 10) {
medewitt marked this conversation as resolved.
    obs <- enw_filter_report_dates(x, remove_days = filter_report_remove)
    obs <- enw_filter_reference_dates(obs, include_days = filter_reference_include)
    return(obs)
}

```

As many of these pre-processing steps are likely to be chained together in repeatable pipelines, there might be an opportunity to provide users with a consolidated helper function.

**Describe the solution you'd like**

A series of helper functions which chain the pre-processing steps for input data that are often the most repeated with sensible defaults.

**Describe alternatives you've considered**
Leaving the pre-processing steps discrete allows for easier debugging should an error occur and potentially less code burden to maintain. 

However, users will likely create these helper functions on their own and providing some streamlined defaults could help with package adoption and ease in productionalising. 

**Additional context**
N/A

**Comment by seabbs** [comment]
Thanks for this @medewitt I think there are definitely positive and negatives (which you've captured). @Bisaloo do you have a strong view on this?

**Comment by Bisaloo** [comment]
>  @Bisaloo do you have a strong view on this?

No, no strong view. It's slightly more maintenance but the small improvement in workflows may be worth it. It can be added as a test and its usefulness can be reevaluated later down the line.

If possible, I would recommend to try and be consistent in your usage in examples & vignettes. Promoting two ways to do the same task can be confusing for new users and being clear on what is the "officially recommended" workflow can help.

**Comment by seabbs** [comment]
Thanks, good points. 

> recommend to try and be consistent in your usage in examples & vignettes.

Yes definitely. 

The only downside I see for this is if we exposed all arguments for the inner two functions that the wrapper would have quite a few.

We might need to workshop the name a bit as well as its a bit vague maybe?

**Comment by seabbs** [comment]
@medewitt any chance you've got capacity to take this one?

**Comment by seabbs** [comment]
Yeah that would be no problem - amazing! Good luck  getting through the swamp!
